src/api/core/tokenManager.spec.ts

```diff
@@ -1,170 +1,286 @@
-      TokenStorage.getStorageAPI().clear();
-      tm = new TokenManager(
-        requestAdapterMockFactory().succesfulExecution({ token: "validToken", refresh_token: "validRefreshToken" }),
-        new Logger()
-      );
+    it("should return a token if authorized by alias and get default token", async () => {
+      const { subject, validOptions, tokens } = createSubject();
+      await subject.init({ ...validOptions, auth: faker.random.word() });
+      expect(await subject.token()).toEqual(tokens.access_token);
     });
+  });
 
+  describe("when the client is terminated", () => {
     it("should have clear the session storage", async () => {
-      spyOn(tm["storage"], "clear");
-      try {
-        await tm.init({ projectID: validProjectID, key: "validKey", secret: "validSecret" });
-        await tm.terminate();
+      const { subject, validOptions } = createSubject();
+      const storage = (subject as any).storage;
+      const clear = storage.clear.bind(storage);
+      spyOn(storage, "clear");
+      await subject.init(validOptions);
+      await subject.terminate();
+      expect(storage.clear).toHaveBeenCalledWith();
+      clear();
+    });
 
-        expect(tm["storage"]["clear"]).toHaveBeenCalledWith();
-      } catch (error) {
-        expect(error).not.toBeDefined();
-      }
+    it("should clear all refreshing intervals", async () => {
+      const { subject, validOptions } = createSubject();
+      const clearIntervalSpy = jest.spyOn(global, "clearInterval");
+      await subject.init(validOptions);
+      await subject.addTokens(faker.random.word(), {
+        access_token: faker.random.word(),
+        refresh_token: faker.random.word(),
+      });
+      const [ firstInterval, secondInterval ] = (subject as any).refreshes;
+      await subject.terminate();
+      // @ts-ignore
+      expect(global.clearInterval).toHaveBeenNthCalledWith(1, firstInterval);
+      // @ts-ignore
+      expect(global.clearInterval).toHaveBeenNthCalledWith(2, secondInterval);
+      expect((subject as any).refreshes).toEqual([]);
+      clearIntervalSpy.mockRestore();
     });
 
     it("should throw an error if the token is accessed after terminate", async () => {
+      const { subject, validOptions } = createSubject();
+      await subject.init(validOptions);
+      await subject.terminate();
       try {
-        await tm.init({ projectID: validProjectID, key: "validKey", secret: "validSecret" });
-        tm.terminate();
-        await tm.token();
-
-        expect(global.clearInterval).toHaveBeenCalledWith(tm["refreshInterval"]);
+        await subject.token();
+        throw new Error("token() should have failed!");
       } catch (error) {
         expect(error.message).toEqual(MESSAGE.TokenManager.TOKEN_NOT_AVAILABLE);
       }
     });
   });
 
   describe("when refreshing the token", () => {
-    let tm: TokenManager;
+    describe("when time for refreshing has come", () => {
+      beforeEach(() => jest.useFakeTimers());
```

Comment:
I'd use this for all tests so they can run faster (or at least the ones that contain async calls)

Reply:
I'm not sure it influences somehow on the tests speed, but done ;)